src/internal/kopia/backup_bases.go

```diff
@@ -180,9 +187,10 @@ func (bb *backupBases) MergeBackupBases(
 	}
 
 	res := &backupBases{
-		backups:     bb.Backups(),
-		mergeBases:  bb.MergeBases(),
-		assistBases: bb.AssistBases(),
+		backups:       bb.Backups(),
+		mergeBases:    bb.MergeBases(),
+		assistBases:   bb.AssistBases(),
+		assistBackups: bb.AssistBackups(),
```

Comment:
I think the code below in this function (can't comment directly) needs some touchups for selecting assist backups too. The code will be similar to lines 189/197-209/217 except we'll only add to assist. I guess we should also be adding the merge backups to the assist backup set? Or do we need to completely separate the assist set and the merge set?

Reply:
looking a this again the code does need expanded to handle the assist backup models. Otherwise we'll accidentally drop them from the set when this function is called

Reply:
Thanks! So I thought about this some more, and removed some migration code which merges legacy assist bases. 

**Summary:**
1. Bases tagged with `fallbackReasons` can only be merge bases, not assist bases. One exception being merge base reported as an assist base. This will be fixed in #3943 
2. Assist backups can never exist for `fallbackReasons`, because we don't do UPN based tagging anymore.

Impact of 1) is minimal. This would impact below scenario.

A) User updates corso to latest, and has a legacy merge base + legacy assist base. Delta token will be sourced from legacy merge base. Assuming its valid, items will be enumerated & due to 1) those cached items will be redownloaded by kopia from graph. 
B) Similar to A) but only has a legacy assist base. We'll do a full backup in this case.